rosidl_typesupport_connext_c/cmake/rosidl_typesupport_connext_c_generate_interfaces.cmake

```diff
@@ -267,3 +257,30 @@ if(NOT rosidl_generate_interfaces_SKIP_INSTALL)
 
   ament_export_libraries(${rosidl_generate_interfaces_TARGET}${_target_suffix} ${Connext_LIBRARIES})
 endif()
+
+if(NOT "${_generated_msg_files}${_generated_srv_files} " STREQUAL " ")
+  find_package(ament_cmake_cppcheck)
```

Comment:
I was speaking with @wjwwood about this. what I understood as the resulting suggestion was:
- add a check for BUILD_TESTING, as previously suggested
- find linters with QUIET, as previously suggested
- add a warning that the tests are being skipped in the event that the linter isn't found (in favour of the generic 'Could not find a package configuration file' warning suppressed above)
- add the appropriate dependencies across our packages to ensure the tests run


(I've opened https://github.com/ros2/rosidl/pull/142 with these changes applied retroactively for https://github.com/ros2/rosidl/commit/454d8d5c70ed54670efe8922d4bd4243f1b0e47a)